injected/src/features/api-manipulation.js

```diff
@@ -0,0 +1,118 @@
+import ContentFeature from '../content-feature';
+import { processAttr } from '../utils';
+
+export default class ApiManipulation extends ContentFeature {
+    init() {
+        const apiChanges = this.getFeatureSetting('apiChanges');
+        if (apiChanges) {
+            for (const scope in apiChanges) {
+                const change = apiChanges[scope];
+                if (!this.checkIsValidAPIChange(change)) {
+                    continue;
+                }
+                this.applyApiChange(scope, change);
+            }
+        }
+    }
+
+    /**
+     * Checks if the config API change is valid.
+     * @param {any} change
+     * @returns {change is APIChange}
+     */
+    checkIsValidAPIChange(change) {
```

Comment:
This basic validation is to prevent a broken config. I'm intending to get this generated from TS used in the config repo when we make that change. This will prevent errors in the config ever happening and also ensure no typing issues on our code side.

Reply:
IIRC there's similar checking that fails safe on the `processAttr` method but I'll check.

Reply:
Do we really need runtime validation for config? It feels like we could just get away with static checks on the config side and avoid the runtime tax?

Reply:
I'm normally of the same opinion too. For this I wanted to block on us adding TS to the config repo before we harden this up further rather than JSON schema it and then rewrite.

Given the risk of misconfig I think the cost here isn't significant and worthwhile to keep at least until we have that hardening.

Reply:
We agreed to add a removal notice task with a timeline in Asana.